### PR TITLE
on error callback function

### PR DESCRIPTION
**Pull request by bo-dun-1**
upfront's request: https://feathery.slack.com/archives/C02AYGFN03Y/p1641355705019800?thread_ts=1641330172.018800&cid=C02AYGFN03Y

tested via hosted-forms and my own onError callback function

### CODE REVIEW

File: src/form/Form.jsx [hunk 1]
```diff
@@ -372,6 +374,31 @@ function Form({
...
+      const values = convertFilesToFilePromises(userVals, fileServarKeys);
+      updateFieldValues(values);
+      client.submitCustom(values);
+    },
+    setProgress: (val) => setUserProgress(val),
+    setStep: (stepKey) => {
+      changeStep(stepKey, activeStep.key, steps, history);
+    },
+    userId: initInfo().userKey,
+    stepName: activeStep?.key || ''
```
Comment (jakeandbakelindvall) [inline src/form/Form.jsx:389]: probably overkill but could do `??` to get into habit of safety it gives

File: src/utils/formHelperFunctions.js [hunk 2]
```diff
@@ -311,13 +314,22 @@ function setFormElementError({
...
     }
     if (triggerErrors) formRef.current.reportValidity();
-    return !formRef.current.checkValidity();
+    invalid = !formRef.current.checkValidity();
   } else if (errorType === 'inline') {
     if (fieldKey) inlineErrors[fieldKey] = { message };
     if (triggerErrors)
       setInlineErrors(JSON.parse(JSON.stringify(inlineErrors)));
-    return Object.values(inlineErrors).find((data) => data.message);
+    invalid = Object.values(inlineErrors).find((data) => data.message);
```
Comment (jakeandbakelindvall) [inline src/utils/formHelperFunctions.js:322]: is this going to be setting `invalid` to something other than a boolean? seems odd if so, but I also could be missing how the shape of `inlineErrors` actually goes

**Review by jakeandbakelindvall — COMMENTED** [review]
2 small things

File: src/utils/formHelperFunctions.js (hunk 2 above)
Comment (bo-dun-1) [reply src/utils/formHelperFunctions.js]: now coercing to a boolean

**Review by jakeandbakelindvall — APPROVED** [review]

**Pull request merged**